agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -1561,6 +1572,12 @@ void ap_manager_thread::handle_hostapd_attached()
     // Copy the channels supported by the AP
     copy_radio_supported_channels(ap_wlan_hal, notification->params().supported_channels);
 
+    auto supported_channels_list = &std::get<1>(notification->supported_channels_list(0));
+    for (uint i = 0, j = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
+        if (hw_supported_channels[i].channel != 0)
+            supported_channels_list[j++] = hw_supported_channels[i];
+    }
```

Comment:
This is silly... we first copy into a temporary array and then copy again into another array. Why not simply copy from ap_wlan_hal here?
```suggestion
    uint i = 0;
    for (const auto &channel : ap_wlan_hal->get_radio_info().supported_channels) {
        if (channel.channel != 0) {
            supported_channels[i].channel        = channel.channel;
            ...
        }
    }
```

You could refactor that into a second copy_supported_channels function since it will probably be used elsewhere as well.

Reply:
the point is moot,
The supported_channels are now stored as part of the RadioInfo struct